service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java

```diff
@@ -791,6 +801,28 @@ public ResponseEntity<ApiAzureManagedIdentityResource> getAzureManagedIdentity(
         fetchCreateControlledKubernetesNamespaceResult(jobId), HttpStatus.OK);
   }
 
+  /**
+   * Initial implementations of createAzureKubernetesNamespace and createAzureDatabase required ids
+   * of the resources to be passed in. But names are easier to use and port nicely during clone
+   * operations. So currently the api supports both names and ids. But we convert to names before
+   * storing the resource.
```

Comment:
Is this going to create a change in query response type (from ID to name) too? Is that going to break anything downstream?

Reply:
I merged [this PR](https://github.com/DataBiosphere/terra-workspace-manager/pull/1470) to have a phased rollout and avoid a breaking change. I don't think we should use ids at all so I would like to eventually remove that option. The problem comes with cloning resources because the id is not portable across clones but the name is.